presto-hive/src/main/java/com/facebook/presto/hive/ParquetHiveRecordCursor.java

```diff
@@ -707,10 +736,9 @@ public void end()
             for (BlockConverter converter : converters) {
                 converter.afterValue();
             }
-            currentBuilder.getEncoding().writeBlock(out, currentBuilder.build());
-
-            if (builder != null) {
-                VARBINARY.writeSlice(builder, out.copySlice());
+            builder.closeEntry();
+            if (needToProduceResultBlock) {
+                resultBlock = builder.getObject(0, Block.class);
```

Comment:
also, if you store the `prestoType` you got in the constructor, then you can use `type.getObject` which would be more correct